src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallDocumentService.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallDocument
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallDocumentCategory
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallRepository
+import uk.gov.justice.digital.hmpps.managerecallsapi.storage.S3Service
+import java.util.UUID
+import javax.persistence.EntityNotFoundException
+
+@Service
+class RecallDocumentService(
+  @Autowired private val s3Service: S3Service,
+  @Autowired private val recallRepository: RecallRepository
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun addDocumentToRecall(recallId: UUID, documentBytes: ByteArray, documentCategory: RecallDocumentCategory) {
+    val recall = try {
+      recallRepository.getById(recallId)
+    } catch (e: EntityNotFoundException) {
+      throw RecallNotFoundError("No recall found with ID '$recallId'", e)
```

Comment:
Could think about having a service level exception handler for EntityNotFoundExceptions, I'm guessing the service should always respond with 404 if this happens for any missing Entity.  Then we don't need specific exception logic for every  entity

Reply:
Isn't it better to throw a specific exception rather than a Hibernate one? I think it gives us more control. Then we can have a RecallNotFoundError handler which responds with a 404 if necessary? In this case I don't think you would respond with a 404 when you're creating a document. Probably just a 400, i.e. the operation that you are attempting is not valid since the recall doesn't exist.

Reply:
We're kind of tied to Hibernate so I don't mind having a specific handler for that exception.  Plus it would then cater for any new hibernate types, otherwise we'll need to catch for every future repository and add a new Exception type.  Just thinking of ways to reduce the code and simplify.  We don't need to worry too much about it now as we only have one repository, just need to make sure this error is handled correctly

Something not found to me should be a 404.  A 400 signifies there's something invalid about the request you sent, it wasn't an invalid request, just the request was being made for something didn't exist

Reply:
We're talking about code that doesn't exist but:
REST is about resources 
404 means "resource not found"
If we add an endpoint for `POST /recalls/{recallId}/documents` the resource is "documents"
In this case the document resource has not even been created so responding with 404 is does not make sense from a REST point of view, or at least would be very confusing

Reply:
Exactly

404 means "resource not found"

The documents can not be found, therefore 404.  Makes perfect sense to me, not sure what is confusing about it

Reply:
It's not the document that hasn't been found, it's the recall ; the document doesn't exist yet.

Reply:
Ok, so we would be returning 404 because the Recall was not found, because it does not exist, same as the documents.

Under what circumstances would you return a 404?